src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -1218,6 +1359,44 @@ private bool TryBindProperty(string identifier, out IEdmProperty projectedProper
             return projectedProperty != null;
         }
 
+        private bool TryBindEscapeFunction()
+        {
+            ODataPathSegment previous = null;
+            if (this.parsedSegments.Count > 0)
+            {
+                previous = this.parsedSegments[this.parsedSegments.Count - 1];
+            }
+
+            IEdmType bindingType = null;
+            if (previous != null)
+            {
+                bindingType = previous.EdmType;
+            }
+
+            string newIdentifier, newParenthesisExpression;
+            bool anotherEscapeFunctionStarts = false;
+
+            if (this.TryResolveEscapeFunction(bindingType, configuration.Model, out newIdentifier, out newParenthesisExpression, out anotherEscapeFunctionStarts))

```

Comment:
this.configuration.Model, to be clear where configuration is coming from?  Note that you don't really have to pass this, since TryResolveEscapeFunction is an instance method and has access to the model through the same configuration property.